### PR TITLE
ADCM-2464 add tests for permissions deleting variants

### PR DESCRIPTION
**Pull request by skhomuti**

### CODE REVIEW

File: tests/functional/rbac/test_permissions_deleting.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+def test_remove_user_from_policy(user_sdk: ADCMClient, user, prepare_objects, sdk_client_fs):
+    """
+    Test that user loses access if user removed from policy user list
+    """
+    cluster_via_admin, *_ = prepare_objects
+    cluster = user_sdk.cluster(id=cluster_via_admin.id)
+    policy = create_policy(sdk_client_fs, BusinessRoles.ViewConfigurations, objects=[cluster], users=[user], groups=[])
+    is_allowed(cluster, BusinessRoles.ViewConfigurations)
+    policy.update(user=[{"id": sdk_client_fs.user(username="admin").id}])
```
Comment (annademina) [inline tests/functional/rbac/test_permissions_deleting.py:27]: allure step?

**Review by annademina — APPROVED** [review]

File: tests/functional/rbac/test_permissions_deleting.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        sdk_client_fs, BusinessRoles.ViewConfigurations, objects=[cluster], users=[user], groups=[group]
+    )
+    is_allowed(cluster, BusinessRoles.ViewConfigurations)
+    policy.update(user=[{"id": sdk_client_fs.user(username="admin").id}])
+    is_allowed(cluster, BusinessRoles.ViewConfigurations)
+
+
+def test_remove_user_from_group_but_still_in_another_group(user_sdk: ADCMClient, user, prepare_objects, sdk_client_fs):
+    """
+    Test that user is still have access if user removed from group but is in another group
```
Comment (Sealwing) [inline tests/functional/rbac/test_permissions_deleting.py:109]: From the test body, it looks like the user isn't removed from the group. The group is removed from the policy. Please, clarify

File: tests/functional/rbac/test_permissions_deleting.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    """
+    cluster_via_admin, service_via_admin, *_ = prepare_objects
+    cluster = user_sdk.cluster(id=cluster_via_admin.id)
+    service = user_sdk.service(id=service_via_admin.id)
+    policy = create_policy(
+        sdk_client_fs, BusinessRoles.ViewConfigurations, objects=[cluster, service], users=[user], groups=[]
+    )
+    is_allowed(service, BusinessRoles.ViewConfigurations)
+    policy.update(object=[{"id": cluster.id, "type": "cluster"}])
+    is_allowed(cluster, BusinessRoles.ViewConfigurations)
```
Comment (Sealwing) [inline tests/functional/rbac/test_permissions_deleting.py:135]: Maybe check that view config of service is allowed?

File: tests/functional/rbac/test_permissions_deleting.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    Test that user is still have access if removed policy but we still have policy with the same rights
+    """
+    cluster_via_admin, *_ = prepare_objects
+    cluster = user_sdk.cluster(id=cluster_via_admin.id)
+    policy = create_policy(sdk_client_fs, BusinessRoles.ViewConfigurations, objects=[cluster], users=[user], groups=[])
+    second_policy = create_policy(
+        sdk_client_fs, BusinessRoles.ViewConfigurations, objects=[cluster], users=[user], groups=[]
+    )
+    is_allowed(cluster, BusinessRoles.ViewConfigurations)
+    policy.update(object=[{"id": cluster.id, "type": "cluster"}])
```
Comment (Sealwing) [inline tests/functional/rbac/test_permissions_deleting.py:149]: What should this step do? It looks like policy already have this cluster in objects

File: tests/functional/rbac/test_permissions_deleting.py [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+def test_remove_object_from_policy(user_sdk: ADCMClient, user, prepare_objects, sdk_client_fs):
+    """
+    Test that user loses access if object changed from policy
+    """
+    cluster_via_admin, service_via_admin, *_ = prepare_objects
+    cluster = user_sdk.cluster(id=cluster_via_admin.id)
+    policy = create_policy(sdk_client_fs, BusinessRoles.ViewConfigurations, objects=[cluster], users=[user], groups=[])
+    is_allowed(cluster, BusinessRoles.ViewConfigurations)
+    policy.update(object=[{"id": service_via_admin.id, "type": "service"}])
+    is_denied(cluster, BusinessRoles.ViewConfigurations)
```
Comment (Sealwing) [inline tests/functional/rbac/test_permissions_deleting.py:69]: Maybe add a check that view service config is now available? It looks like possible problem with reassignment of permissions

**Review by Sealwing — CHANGES_REQUESTED** [review]

File: tests/functional/rbac/test_permissions_deleting.py (hunk 4 above)
Comment (skhomuti) [reply tests/functional/rbac/test_permissions_deleting.py:149]: remove service from objects and check that access still allowed

Comment (Sealwing) [reply tests/functional/rbac/test_permissions_deleting.py:149]: but you added only cluster object up here, no?

File: tests/functional/rbac/test_permissions_deleting.py (hunk 5 above)
Comment (skhomuti) [reply tests/functional/rbac/test_permissions_deleting.py:69]: fixed

File: tests/functional/rbac/test_permissions_deleting.py (hunk 3 above)
Comment (skhomuti) [reply tests/functional/rbac/test_permissions_deleting.py:135]: fixed

File: tests/functional/rbac/test_permissions_deleting.py (hunk 2 above)
Comment (skhomuti) [reply tests/functional/rbac/test_permissions_deleting.py:109]: fixed

File: tests/functional/rbac/test_permissions_deleting.py (hunk 1 above)
Comment (skhomuti) [reply tests/functional/rbac/test_permissions_deleting.py:27]: fixed

File: tests/functional/rbac/test_permissions_deleting.py (hunk 4 above)
Comment (skhomuti) [reply tests/functional/rbac/test_permissions_deleting.py]: cluster rights allow the same permission for services and components too

**Review by Sealwing — APPROVED** [review]

**Pull request merged**